### PR TITLE
Document adding custom button icons for standalone plugins.

### PR DESCRIPTION
**Pull request by lewiscollard**
The previously [recommended](https://github.com/Alex-D/Trumbowyg/issues/1119#issuecomment-631452561) way of adding a custom icon to a plugin's button was to roll one's own build of Trumbowyg, but for lazy people like me, that looks too much like effort.

Moreover, that implies that plugins _cannot_ be distributed standalone (e.g. load Trumbowyg from a CDN, then your own plugin on top of that). This is not so! But that notion might put someone off using Trumbowyg for their project. Trumbowyg doesn't actually care where the button's symbol comes from; if a symbol exists in the document with the correct ID it will work. So, this PR documents the "shove an SVG into the DOM" solution.

P.S. Thanks for Trumbowyg. It's awesome!

### CODE REVIEW

File: docs/documentation/plugins/index.html [hunk 1]
```diff
@@ -1482,6 +1482,44 @@ <h3 id="skeleton">Skeleton</h3>
...
+            const iconWrap = $(document.createElementNS("http://www.w3.org/2000/svg", "svg"));
+
+            /*
+            This class is only here to allow CSS to hide it. In your plugin's
+            stylesheet you will probably want something like this:
+
+            .trumbowyg-myplugin-icons {
+                display: none;
+            }
+            */
```
Comment (Alex-D) [inline docs/documentation/plugins/index.html:1510]: Maybe I should add a generic class like that to the trumbowyg CSS file, allowing it to be shared by the default SVG *and* all additional ones :)

File: docs/documentation/plugins/index.html [hunk 2]
```diff
@@ -1482,6 +1482,44 @@ <h3 id="skeleton">Skeleton</h3>
         }
     }
 
+    // If the plugin is a button
+    function buildButtonIcon() {
+        if (!document.querySelector("#trumbowyg-myplugin")) {
```
Comment (Alex-D) [inline docs/documentation/plugins/index.html:1487]: I prefer the early return way, aka if the element exists, return; then, the rest of the function runs only if we need to create the button. It prevent from having the whole function in a if

**Review by Alex-D — CHANGES_REQUESTED** [review]

File: docs/documentation/plugins/index.html [hunk 3]
```diff
@@ -1482,6 +1482,44 @@ <h3 id="skeleton">Skeleton</h3>
...
+            This class is only here to allow CSS to hide it. In your plugin's
+            stylesheet you will probably want something like this:
+
+            .trumbowyg-myplugin-icons {
+                display: none;
+            }
+            */
+            iconWrap.addClass("trumbowyg-myplugin-icons");
+            // For demonstration purposes, we've taken the "File" icon from
+            // Remix Icon - https://remixicon.com/
```
Comment (Alex-D) [inline docs/documentation/plugins/index.html:1513]: Nice one, did not know it :)

File: docs/documentation/plugins/index.html (hunk 2 above)
Comment (lewiscollard) [reply docs/documentation/plugins/index.html]: Now that you mention it, so do I. See: fa7ead9

File: docs/documentation/plugins/index.html (hunk 1 above)
Comment (lewiscollard) [reply docs/documentation/plugins/index.html]: I think that is a good idea. Is it a blocker for this PR?

**Review by Alex-D — APPROVED** [review]

**Comment by Alex-D** [comment]
Thank you for your suggestion :)

**Pull request merged**